mathics/core/expression.py

```diff
@@ -739,15 +735,25 @@ def equal2(self, rhs: Any) -> Optional[bool]:
         """
         if self.sameQ(rhs):
             return True
+        # if rhs is an Atom, return None
+        elif isinstance(rhs, Atom):
+            return None
 
-        if self.has_form("List", None) and rhs.has_form("List", None):
+        # Here we only need to deal with Expressions.
+        equal_heads = self._head.equal2(rhs._head)
+        if not equal_heads:
+            return equal_heads
+        # From here, we can assume that both heads are the same
+        if self.get_head_name() in ("System`List", "System`Sequence"):
             if len(self._leaves) != len(rhs._leaves):
                 return False
             for item1, item2 in zip(self._leaves, rhs._leaves):
                 result = item1.equal2(item2)
                 if not result:
                     return result
             return True
+        elif self.get_head_name() in ("System`DirectedInfinity",):
```

Comment:
I wonder if there isn't way to add `equal2` to `DirectedInfinity` so this code appears elsewhere? 

What do you think?

Reply:
See https://github.com/mathics/Mathics/pull/1273/files#r612879671